manifests/certonly.pp

```diff
@@ -252,7 +252,6 @@
     environment => $environment,
     provider    => 'shell',
     require     => [
-      Class['letsencrypt'],
```

Comment:
Isn't this still needed for certbot to installed and initialized?

Reply:
This exec is the correct one for now. It would also be great if there was a test that validates all dependencies are correct. Dependency cycles are easy to detect in CI.

It looks like this code here checks the parameter:
https://github.com/voxpupuli/puppet-letsencrypt/blob/0eaecd43b1f7c588b5c1373a694bb4ce6fde801c/spec/classes/letsencrypt_spec.rb#L146-L156
Please add `it { is_expected.to compile.with_all_deps }` to it. I'd expect that to fail in master and your commit to solve it. Then we have a nice regression test.

Reply:
done: https://github.com/voxpupuli/puppet-letsencrypt/pull/275/commits/e81f72210225ac3bf6b3f00cc436c52373c05065